dockerplugin.py

```diff
@@ -412,10 +463,79 @@ def read_callback(self):
                     if value:
                         method(container, cstats.dimensions, value, read_at)
             except Exception, e:
-                collectd.warning(('Error getting stats for container '
-                                  '{container}: {msg}')
-                                 .format(container=_c(container), msg=e))
-                collectd.warning(traceback.format_exc())
+                log.warning(('Unable to retrieve stats for container '
+                             '{container}: {msg}')
+                            .format(container=_c(container), msg=e))
+                # Print stack trace to log if verbose logging is on
+                log.info(traceback.format_exc())
+
+
+class Log:
```

Comment:
I'd rather you implement a `logging.Handler` and use the rest of the logging module's functionality to wire this together.